odoo/fields.py

```diff
@@ -2376,6 +2376,32 @@ def __init__(self, comodel_name=Default, relation=Default, column1=Default,
             **kwargs
         )
 
+    def _get_mro_has_field_relation(self, model, name):
+        for mro in type(model).__mro__:
+            if hasattr(mro, '_abstract') and\
+                    not mro._abstract and name in mro.__dict__:
+                field = mro.__dict__[name]
+                return self.relation == field.args.get('relation')
+        return False
+
+    def _check_relation(self, model, name):
+        if isinstance(model._inherit, str):
+            inheriteds = [model._inherit]
+        elif isinstance(model._inherit, list):
+            inheriteds = model._inherit
+        else:
+            return
+        for inherited in inheriteds:
+            inherited_model = model.env[inherited]
+            if model._name != inherited:
+                if self._get_mro_has_field_relation(inherited_model, name):
+                    self.relation = False
```

Comment:
why False?

Reply:
That's the only point that solves all!
This is sufficent to let https://github.com/odoo/odoo/blob/07c111fa7255b28e901c21f2f9967bda0422ab8a/odoo/fields.py#L2385 calculate the relation value

Reply:
NICE